### ISSUE TITLE
Understanding ensembling

### ISSUE THREAD

**Issue by MrRobot2211** [pudae/kaggle-understanding-clouds#8]
Hi excellent contribution. I am trying to understand how your model ensembling is implemented but I can not find the definitions of the hook methods in evaluate.py .

outputs = hooks.post_forward_fn(outputs=outputs, images=images, data=data, is_train=False)
loss = hooks.loss_fn(outputs=outputs, labels=labels, data=data, is_train=False)

Are this the functions responsible for the ensembling?

**Comment by pudae** [comment]
Hi, MrRobot2211
Thank you for your interest.
Ensembling is implemented in tools/evaluate.py and tools/make_submission.py
